library/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,623 @@
+            origin_path=self.origin_path,
+            content_types_to_compress=default_content_types() if self.is_compression_enabled and not self.content_types_to_compress else self.content_types_to_compress,
+            is_compression_enabled=self.is_compression_enabled if self.is_compression_enabled is not None else False,
+            is_http_allowed=self.is_http_allowed if self.is_http_allowed is not None else True,
+            is_https_allowed=self.is_https_allowed if self.is_https_allowed is not None else True,
+            query_string_caching_behavior=self.query_string_caching_behavior if self.query_string_caching_behavior else QueryStringCachingBehavior.ignore_query_string
+        )
+
+        try:
+            poller = self.cdn_management_client.endpoints.create(self.resource_group, self.profile_name, self.name, parameters)
+            response = self.get_poller_result(poller)
+            return cdnendpoint_to_dict(response)
+        except ErrorResponseException as exc:
+            self.log('Error attempting to create Azure CDN endpoint instance.')
+            self.fail("Error creating Azure CDN endpoint instance: {0}".format(exc.message))
+
+    def update_cdnendpoint(self):
+        '''
+        Updates a Azure CDN endpoint.
+
+        :return: deserialized Azure CDN endpoint instance state dictionary
+        '''
+        self.log("Updating the Azure CDN endpoint instance {0}".format(self.name))
+
+        endpoint_update_properties = EndpointUpdateParameters(
+            tags=self.tags,
+            origin_host_header=self.origin_host_header,
+            origin_path=self.origin_path,
+            content_types_to_compress=default_content_types() if self.is_compression_enabled and not self.content_types_to_compress else self.content_types_to_compress,
+            is_compression_enabled=self.is_compression_enabled,
+            is_http_allowed=self.is_http_allowed,
+            is_https_allowed=self.is_https_allowed,
+            query_string_caching_behavior=self.query_string_caching_behavior,
+        )
+
+        try:
+            poller = self.cdn_management_client.endpoints.update(self.resource_group, self.profile_name, self.name, endpoint_update_properties)
+            response = self.get_poller_result(poller)
+            return cdnendpoint_to_dict(response)
+        except ErrorResponseException as exc:
+            self.log('Error attempting to update Azure CDN endpoint instance.')
+            self.fail("Error updating Azure CDN endpoint instance: {0}".format(exc.message))
+
+    def delete_cdnendpoint(self):
+        '''
+        Deletes the specified Azure CDN endpoint in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the CDN endpoint {0}".format(self.name))
+        try:
+            poller = self.cdn_management_client.endpoints.delete(
+                self.resource_group, self.profile_name, self.name)
```

Comment:
a side question, if delete cdn profile, will cdn endpoints be deleted?

Reply:
Yes. Endpoints are affiliated to a specific CDN profile. Profile is similar to a resource group, once deleted, all endpoints in that profile are gone. (Tested in portal and Ansible)